tests/e2e/validatecluster/validatecluster_test.go

```diff
@@ -1,73 +1,257 @@
+var nodeOS = flag.String("nodeOS", "generic/ubuntu2004", "a string")
+var serverCount = flag.Int("serverCount", 3, "an int")
+var agentCount = flag.Int("agentCount", 2, "an int")
+
+//valid format: RELEASE_VERSION=v1.23.1+k3s2 or nil for commit
+var installType = flag.String("installType", "", "a string")
+
 func Test_E2EClusterValidation(t *testing.T) {
 	RegisterFailHandler(Fail)
 	RunSpecsWithDefaultAndCustomReporters(t, "Create Cluster Test Suite", []Reporter{
 		reporters.NewJUnitReporter("/tmp/results/junit-cc.xml"),
 	})
 }
 
-const (
-	// Valid nodeOS: generic/ubuntu2004, opensuse/Leap-15.3.x86_64, dweomer/microos.amd64
-	nodeOs      = "generic/ubuntu2004"
-	serverCount = 3
-	agentCount  = 2
-)
-
 var (
 	kubeConfigFile  string
 	serverNodeNames []string
 	agentNodeNames  []string
 )
 
-var _ = Describe("Verify Cluster Creation", func() {
-	Context("Create the Cluster", func() {
+var _ = Describe("Verify Create", func() {
+	Context("Cluster :", func() {
 		It("Starts up with no issues", func() {
 			var err error
-			serverNodeNames, agentNodeNames, err = e2e.CreateCluster(nodeOs, serverCount, agentCount)
+			serverNodeNames, agentNodeNames, err = e2e.CreateCluster(*nodeOS, *serverCount, *agentCount, *installType)
 			Expect(err).NotTo(HaveOccurred())
 			fmt.Println("CLUSTER CONFIG")
-			fmt.Println("OS:", nodeOs)
+			fmt.Println("OS:", *nodeOS)
 			fmt.Println("Server Nodes:", serverNodeNames)
 			fmt.Println("Agent Nodes:", agentNodeNames)
 			kubeConfigFile, err = e2e.GenKubeConfigFile(serverNodeNames[0])
 			Expect(err).NotTo(HaveOccurred())
 		})
-		It("Verify Node and Pod Status", func() {
+
+		It("Node and Pod Status", func() {
 			fmt.Printf("\nFetching node status\n")
 			Eventually(func(g Gomega) {
-				nodes, err := e2e.ParseNodes(kubeConfigFile, true)
+				nodes, err := e2e.ParseNodes(kubeConfigFile, false)
 				g.Expect(err).NotTo(HaveOccurred())
 				for _, node := range nodes {
-					g.Expect(node.Status).Should(Equal("Ready"), func() string { return node.Name })
+					g.Expect(node.Status).Should(Equal("Ready"))
 				}
 			}, "420s", "5s").Should(Succeed())
+			_, _ = e2e.ParseNodes(kubeConfigFile, true)
 
 			fmt.Printf("\nFetching Pods status\n")
 			Eventually(func(g Gomega) {
-				pods, err := e2e.ParsePods(kubeConfigFile, true)
+				pods, err := e2e.ParsePods(kubeConfigFile, false)
 				g.Expect(err).NotTo(HaveOccurred())
 				for _, pod := range pods {
 					if strings.Contains(pod.Name, "helm-install") {
-						g.Expect(pod.Status).Should(Equal("Completed"), func() string { return pod.Name })
+						g.Expect(pod.Status).Should(Equal("Completed"), pod.Name)
 					} else {
-						g.Expect(pod.Status).Should(Equal("Running"), func() string { return pod.Name })
+						g.Expect(pod.Status).Should(Equal("Running"), pod.Name)
 					}
 				}
 			}, "420s", "5s").Should(Succeed())
+			_, _ = e2e.ParsePods(kubeConfigFile, true)
+		})
+		It("Verify ClusterIP Service", func() {
```

Comment:
Nit: All "It" statements should read as a sentence.
```suggestion
		It("Verifies ClusterIP Service", func() {
```